src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,46 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        String randomColor = String.valueOf(colorSupplier.getRandomColor());
+        Random random = new Random();
+        int figureType = random.nextInt(5);
+
+        switch (figureType) {
+            case 0:
+                double sideLength = random.nextDouble() * 10 + 1;
+                return new Square(randomColor, sideLength);
+
```

Comment:
move it to the private method
```suggestion
                return preapreSquare(randomColor);
```